### PR TITLE
Add support to secrets

### PR DESCRIPTION
**Pull request by drgarcia1986**
There is a problem, change one secret that already exists doesn't perform the rolling update (because the deploy spec doesn't change).

### CODE REVIEW

**Comment by drgarcia1986** [comment]
There's a way to "hack" this problem, according to [this comment](https://github.com/kubernetes/kubernetes/issues/27081#issuecomment-327321981), we can change (or create) an annotation inside of the spec of deploy (_spec.template.metadata.annotations_) with a random value or the current date (doesn't matter).
I made some simple tests and works fine, but, every `env-[un]set` and `secret-[un]set` will perform a rolling update, what do you think guys?

File: pkg/client/cmd/app.go [hunk 1]
```diff
@@ -499,6 +499,153 @@ func appEnvUnset(cmd *cobra.Command, args []string) {
...
+		client.PrintErrorAndExit("Error connecting to server: %s", err)
+	}
+	defer conn.Close()
+
+	cli := appb.NewAppClient(conn)
+	req := &appb.SetEnvRequest{Name: appName, EnvVars: evs}
+	if _, err := cli.SetSecret(context.Background(), req); err != nil {
+		client.PrintErrorAndExit(client.GetErrorMsg(err))
+	}
+	fmt.Println("Env vars updated with success")
```
Comment (aguerra) [inline pkg/client/cmd/app.go]: maybe Secrets updated? As a secret is exposed as env var I don't know which is best

File: pkg/client/cmd/app.go [hunk 2]
```diff
@@ -712,9 +861,15 @@ func init() {
 
 	appEnvSetCmd.Flags().String("app", "", "app name")
 	appEnvSetCmd.Flags().Bool("no-input", false, "set env vars without warning")
-	// App unset env vars
```
Comment (aguerra) [inline pkg/client/cmd/app.go]: :clap:

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,8 +10,9 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ### Added
 - Support Nginx as sidecar
-- support for internal apps
-- support to filter app logs by container name
+- Support for internal apps
+- Support to filter app logs by container name
+- Support to Secrets as env var
```
Comment (aguerra) [inline CHANGELOG.md]: Maybe `Support to set secrets as env vars`

File: pkg/client/cmd/app.go [hunk 4]
```diff
@@ -373,47 +412,20 @@ WARNING:
...
-		tmp := strings.SplitN(item, "=", 2)
-		if len(tmp) != 2 {
-			client.PrintErrorAndExit("Env vars must be in the format FOO=bar")
-		}
-		evs[i] = &appb.SetEnvRequest_EnvVar{Key: tmp[0], Value: tmp[1]}
-	}
-
 	currentClusterName := cfgCluster
 	if currentClusterName == "" {
+		var err error
```
Comment (aguerra) [inline pkg/client/cmd/app.go]: :thinking: why?

**Comment by aguerra** [comment]
For me there's no problem in always doing a rolling update.

File: pkg/client/cmd/app.go (hunk 4 above)
Comment (yagonobre) [reply pkg/client/cmd/app.go]: 🤔

File: pkg/server/k8s/client.go [hunk 5]
```diff
@@ -784,6 +796,35 @@ func (c *Client) patchCronJobEnvVars(namespace, name string, v interface{}) erro
 	return errors.Wrap(err, "patch cronjob failed")
 }
 
+func convertAppSecretEnvVar(secretName string, secrets []string) interface{} {
```
Comment (yagonobre) [inline pkg/server/k8s/client.go]: Maybe put this on `converters.go`

**Comment by yagonobre** [comment]
Agree with @aguerra

File: pkg/server/app/app_test.go [hunk 6]
```diff
@@ -963,6 +989,132 @@ func TestAppOperationsUnsetEnvForACronJob(t *testing.T) {
...
+	user := &database.User{Email: "teresa@luizalabs.com"}
+	app := &App{Name: "teresa", Team: "luizalabs"}
+	tops.(*team.FakeOperations).Storage[app.Team] = &database.Team{
+		Name:  app.Team,
+		Users: []database.User{*user},
+	}
+	ops.(*AppOperations).kops = &errK8sOperations{
+		CreateOrUpdateDeploySecretEnvVarsErr: ErrInvalidEnvVarName,
+		GetSecretErr:                         ErrNotFound,
+		NamespaceErr:                         nil,
```
[the remark this answers is not in the document]
Comment (drgarcia1986) [reply pkg/server/app/app_test.go]: Yes, my fault

File: pkg/client/cmd/app.go (hunk 4 above)
Comment (drgarcia1986) [reply pkg/client/cmd/app.go]: to solve `go build` issue:
```
pkg/client/cmd/app.go|418| 30: currentClusterName declared and not used
```
😢

File: pkg/server/k8s/client.go (hunk 5 above)
Comment (drgarcia1986) [reply pkg/server/k8s/client.go]: I'm not sure, `converters.go` is for _teresa struct_ -> _k8s struct_

**Comment by drgarcia1986** [comment]
Thanks @aguerra and @yagonobre I'll add the "fake" annotation to forces the rolling update

**Comment by drgarcia1986** [comment]
@aguerra @yagonobre UPDATED

**Pull request merged**